### PR TITLE
Design optimizer groups for benchmarks

### PR DESCRIPTION
**Pull request by jrapin**
## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/benchmark/optimizer_groups.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+all_bo = ['BO', 'LBO', 'MidQRBO', 'QRBO', 'RBO', 'chainBOwithLHS', 'chainBOwithLHS30', 'chainBOwithLHSdim', 'chainBOwithLHSsqrt', 'chainBOwithMetaRecentering', 'chainBOwithMetaRecentering30', 'chainBOwithMetaRecenteringdim', 'chainBOwithMetaRecenteringsqrt', 'chainBOwithR', 'chainBOwithR30', 'chainBOwithRdim', 'chainBOwithRsqrt']
```
Comment (jrapin) [inline nevergrad/benchmark/optimizer_groups.txt]: This file is automatically generated to be able to check groups in a glance

File: nevergrad/benchmark/optgroups.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+@registry.register
+def all_bo() -> tp.List[Optim]:
+    return sorted(x for x in ng.optimizers.registry if "BO" in x)
+
+
+@registry.register
+def spsa() -> tp.List[Optim]:
+    # return sorted(x for x, y in ng.optimizers.registry.items() if (any(e in x for e in "TBPSA SPSA".split()) and "iscr" not in x))
+    return ["NaiveTBPSA", "SPSA", "TBPSA"]
```
Comment (jrapin) [inline nevergrad/benchmark/optgroups.py]: both this line and the line above are equivalent, I prefer the second one, it's shorter and easier to read :D

**Comment by teytaud** [comment]
> @teytaud for now I can't figure out a pattern for group of optimizers in the benchmarks, we'll discuss this next week when I am back if you are OK.

sure, no emergency anyway, happy to discuss :-)    all BO variants can be grouped as they have specificities, for the rest it's not that easy... maybe instead of families we can just have indicators, like "is_expensive_in_high_dim" or "is_expensive" or "is_not_parallel" or "does_not_like_noise" or whatever.

**Comment by jrapin** [comment]
> maybe instead of families we can just have indicators, like "is_expensive_in_high_dim" or "is_expensive" or "is_not_parallel" or "does_not_like_noise" or whatever.

If that's what you think of when you build the experiments, then that's exatly what we need. We'll try that at our next meeting if it's ok for you

**Comment by teytaud** [comment]
I liked that idea, but maybe you prefer not to merge it ? or you want me to fix the conflicts as I've been active on experiments.py in the mean time ? no pb for doing this.

**Comment by jrapin** [comment]
I'll update it within the next couple of days if this is fine for you + add a seed for choosing the optimizers order, as we discussed this can solve some of your problems.

File: nevergrad/benchmark/experiments.py [hunk 3]
```diff
@@ -488,16 +490,13 @@ def illcondipara(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
...
-              "MultiCMA", "TripleCMA", "MultiScaleCMA", "RSQP", "RCobyla", "RPowell", "SQPCMA", "MetaModel"]
     functions = [
         ArtificialFunction(name, block_dimension=50, rotation=rotation) for name in ["cigar", "ellipsoid"] for rotation in [True, False]
     ]
-    for optim in optims:
-        for function in functions:
-            for budget in [100, 1000, 10000]:
+    for function in functions:
+        for budget in [100, 1000, 10000]:
+            optims = get_optimizers("large", seed=next(seedg))
```
Comment (jrapin) [inline nevergrad/benchmark/experiments.py:498]: @teytaud with this you should get random order for optimizers

File: nevergrad/benchmark/experiments.py [hunk 4]
```diff
@@ -488,16 +490,13 @@ def illcondipara(seed: tp.Optional[int] = None) -> tp.Iterator[Experiment]:
     """Testing optimizers on ill-conditionned parallel optimization.
     """
     seedg = create_seed_generator(seed)
-    optims = ["NGO", "Shiwa", "DiagonalCMA", "CMA", "PSO", "DE", "MiniDE", "QrDE", "MiniQrDE", "LhsDE", "OnePlusOne",
-              "TwoPointsDE", "OnePointDE", "AlmostRotationInvariantDE", "RotationInvariantDE",
-              "Portfolio", "ASCMADEthird", "ASCMADEQRthird", "ASCMA2PDEthird", "CMandAS2", "CMandAS", "CM",
-              "MultiCMA", "TripleCMA", "MultiScaleCMA", "RSQP", "RCobyla", "RPowell", "SQPCMA", "MetaModel"]
```
Comment (jrapin) [inline nevergrad/benchmark/experiments.py:494]: I've merge this list with the one for constrained_illconditioned_parallel
If that's fine to you can merge the PR

File: nevergrad/benchmark/experiments.py (hunk 3 above)
Comment (teytaud) [reply nevergrad/benchmark/experiments.py:498]: Excellent!

**Comment by teytaud** [comment]
I merge! I might modify some list after my current xps, in particular for reducing, but this is for sure a good step forward, thx.

**Pull request merged**